src/viam/trajex/totg/trajectory.cpp

```diff
@@ -494,8 +508,16 @@ enum class integration_event : std::uint8_t {
 
         if (is_switching_point) {
             // Switching velocity is the minimum (where the discontinuity meets the feasible region)
-            const auto s_dot_switching = std::min(s_dot_max_acc_before, s_dot_max_acc_after);
-            return trajectory::phase_point{.s = boundary, .s_dot = s_dot_switching};
+            const auto s_dot_max_acc_switching_min = std::min(s_dot_max_acc_before, s_dot_max_acc_after);
+
+            // But make sure we don't violate whichever side's velocity limit is constraining.
+            const auto s_dot_max_vel_switching_min = std::min(s_dot_max_vel_before, s_dot_max_vel_after);
+
+            // NOTE: `Divergent Behavior 2`: Only accept this as switching point if it is also feasible with respect to
```

Comment:
We can't return an accel switching point that is in the infeasible region w.r.t. the velocity limit curve.